test/Data.purs

```diff
@@ -123,36 +127,28 @@ suite = do
           test (show input) do
             fromData (toData input) `shouldEqual` Just input
       group "BigInt" do
-        let
-          inputs =
-            [ BigInt.fromInt 0
-            , BigInt.fromInt 10000
-            , BigInt.fromInt (-10000)
-            ]
-        for_ inputs \input -> do
-          test (show input) do
-            fromData (toData input) `shouldEqual` Just input
-      test "Array" do
-        let
-          input = [ Just true, Just false, Nothing ]
-        fromData (toData input) `shouldEqual` Just input
-      test "Map" do
-        let
-          input = Map
-            [ BigInt.fromInt 13 /\
-                [ Map
-                    [ BigInt.fromInt 17 /\ false
-                    ]
-                ]
-            ]
-        fromData (toData input) `shouldEqual` Just input
+        test "property test" $ quickCheck \(n :: Int) ->
+          let
+            input = BigInt.fromInt n
+          in
+            fromData (toData input) === Just input
+        test "Array" do
+          let
+            input = [ Just true, Just false, Nothing ]
+          fromData (toData input) `shouldEqual` Just input
+        test "Map" do
+          let
+            input = Map
+              [ BigInt.fromInt 13 /\
+                  [ Map
+                      [ BigInt.fromInt 17 /\ false
+                      ]
+                  ]
+              ]
+          fromData (toData input) `shouldEqual` Just input
     group "Generic" do
-      -- TODO: Quickcheckify
-      test "EType: from . to == id" do
-        let
-          input = E0 (D1 (D2 (C1 Nothing))) true
-            (C2 (MyBigInt (BigInt.fromInt 123)) false)
-        fromData (toData input) `shouldEqual` Just input
+      test "EType: from . to == id" $ quickCheck \(input :: EType) ->
+        fromData (toData input) === Just input
```

Comment:
Perhaps even refactor the whole property into

```purescript
plutusDataRoundtripProperty :: forall (a :: Type). FromData a => ToData a => Proxy a -> Spec
plutusDataRoundtripProperty (_ :: Proxy a) = quickCheck \(input :: a) -> fromData (toData input) === Just input
```

(P.S didn't check the types myself :))

Reply:
We have 4 or 5 different ways to write tests, so I just abstract most of them into two general functions (one is `plutusDataRoundtripProperty` as suggested )